src/YoutubeAtom.tsx

```diff
@@ -227,13 +227,20 @@ export const YoutubeAtom = ({
                       adTargeting.customParams,
                       consentState,
                   );
-        const embedConfig = encodeURIComponent(JSON.stringify({ adsConfig }));
-        const originString = origin
-            ? `&origin=${encodeURIComponent(origin)}`
-            : '';
-        setIframeSrc(
-            `https://www.youtube.com/embed/${assetId}?embed_config=${embedConfig}&enablejsapi=1&widgetid=1&modestbranding=1${originString}&autoplay=1`,
+
+        const params = new URLSearchParams({
+            ...(origin ? { origin } : {}),
+            autoplay: '1',
+            embed_config: JSON.stringify({ adsConfig }),
+            enablejsapi: '1',
+            modestbranding: '1',
+            widgetid: '1',
+        });
+        const src = new URL(
```

Comment:
this works with 94.79% of current guardian users, are we ok with that number?

Reply:
For some reason I’d thought this would automatically get transpiled by Babel… but actually it’s part of our [polyfill bundle] (comes with `URLSearchParams`).

[polyfill bundle]: https://github.com/guardian/dotcom-rendering/blob/9523184ba928d242f24dbc91c441a7b9d7635aae/dotcom-rendering/src/web/server/document.tsx#L194

I’ve seen it used in:
- [`dotcom-rendering` for the `FilterKeyEventsToggle`](https://github.com/guardian/dotcom-rendering/blob/dc718e5919f04509ebaeca82bdd7ff5c95ddd672/dotcom-rendering/src/web/components/FilterKeyEventsToggle.importable.tsx#L33)
- [`dotcom-rendering` for the `ShadyPie`](https://github.com/guardian/dotcom-rendering/blob/2821e7df597573ffece867163188bb98f2d752b3/dotcom-rendering/src/web/components/ShadyPie.tsx#L1-L11)

Reply:
fantastic 🪂

Reply:
Thanks for your diligence, always!